haystack/document_stores/elasticsearch.py

```diff
@@ -1886,30 +1890,21 @@ def _create_document_index(self, index_name: str, headers: Optional[Dict[str, st
                             f"e.g. `OpenSearchDocumentStore(index='my_new_{self.similarity}_index', similarity='{self.similarity}')`."
                         )
 
-                embedding_field_similarity = self.space_type_to_similarity[embedding_field_space_type]
-                if embedding_field_similarity == self.similarity:
-                    self.embeddings_field_supports_similarity = True
-                else:
-                    if self.similarity == "dot_product":
-                        raise Exception(
-                            f"Embedding field '{self.embedding_field}' is optimized for similarity '{embedding_field_similarity}'. "
-                            f"OpenSearch does not support the use of similarity '{self.similarity}' on '{embedding_field_similarity}'-optimized fields. "
-                            f"In order to try out '{self.similarity}' similarity on this index, you might want to use a different embedding field by setting the `embedding_field` param. "
-                            f"Consider creating a new index optimized for '{self.similarity}' by setting `similarity='{self.similarity}'` the first time you instantiate OpenSearchDocumentStore for the new index, "
-                            f"e.g. `OpenSearchDocumentStore(index='my_new_{self.similarity}_index', similarity='{self.similarity}')`."
-                        )
-                    logger.warning(
-                        f"Embedding field '{self.embedding_field}' is optimized for similarity '{embedding_field_similarity}'. "
-                        f"Falling back to slow exact vector calculation. "
-                        f"Consider creating a new index optimized for '{self.similarity}' by setting `similarity='{self.similarity}'` the first time you instantiate OpenSearchDocumentStore for the new index, "
-                        f"e.g. `OpenSearchDocumentStore(index='my_new_{self.similarity}_index', similarity='{self.similarity}')`."
-                    )
+            # Adjust global ef_search setting. If not set, default is 512.
+            ef_search = index_settings.get("knn.algo_param", {"ef_search": 512}).get("ef_search", 512)
+            if self.index_type == "hnsw" and ef_search != 20:
+                body = {"knn.algo_param.ef_search": 20}
+                self.client.indices.put_settings(index=self.index, body=body, headers=headers)
+            elif self.index_type == "flat" and ef_search != 512:
+                body = {"knn.algo_param.ef_search": 512}
+                self.client.indices.put_settings(index=self.index, body=body, headers=headers)
+
             return
 
         if self.custom_mapping:
-            mapping = self.custom_mapping
+            index_definition = self.custom_mapping
         else:
-            mapping = {
+            index_definition = {
```

Comment:
The `mapping` terminology comes from ElasticSearch, I assume. Let's talk about pros and cons in a call.